### DISCUSSION TITLE
Running librdkafka in a FIPS enabled environment

### DISCUSSION THREAD

**Discussion by navidsh** [edenhill/librdkafka#4343] [Q&A]
Is there anywhere I can find instructions for running librdafka in a FIPS enabled environment?

I found https://github.com/confluentinc/librdkafka/discussions/3702 that mentions librdkafka is not FIPS compliant. But reading through https://github.com/confluentinc/librdkafka/issues/758 it seems like librdkafka can be used in FIPS compliant environments.

**Comment by pranavrth** [comment]
Librdkafka is FIPS compliant when enabled to run in properly configured FIPS environment with OpenSSL 3. Helping instructions are present in [Python client codebase](https://github.com/confluentinc/confluent-kafka-python/tree/master/examples/fips#enabling-fips-provider) though these instructions are mainly for enabling FIPS and are similar for librdkafka as well.

**Reply by kirill-y** [reply]
In the mentioned above python related document, there are two ways of to plug FIPS provider, with precompiled value and environment variables, both or them doesn't meet my needs. I need to control FIPS mode regardless, the file location and the environment values. The only method that I found it to use nondefault ssl fips library context. 
Is it possilbe to "Programmatically loading the FIPS module (nondefault library context)" and use this FIPS context for librdkafka SSL/TLS purposes ?

I mean "Programmatically loading the FIPS module (nondefault library context)" and 
"Using the FIPS module in SSL/TLS" secoion of  [OpenSSL fips module guide](https://www.openssl.org/docs/man3.0/man7/fips_module.html) doc.
